### PR TITLE
Symlink internal dir in e2e

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** symlink both `pkg` and `internal` dirs to the test project's `vendor` when running e2e tests locally, and some cleanup.

**Motivation for the change:** the e2e tests symlink `internal` so any future changes are picked up by local e2e tests.

### CODE REVIEW

File: test/e2e/memcached_test.go [hunk 1]
```diff
@@ -217,17 +217,20 @@ func TestMemcached(t *testing.T) {
...
 
+	// link local sdk to vendor if not in travis
+	if repo == "" {
+		for _, dir := range []string{"pkg", "internal"} {
+			repoDir := filepath.Join("github.com/operator-framework/operator-sdk", dir)
+			vendorDir := filepath.Join("vendor", repoDir)
+			os.RemoveAll(vendorDir)
+			os.Symlink(filepath.Join(gopath, "src", repoDir), vendorDir)
+		}
+	}
```
Comment (AlexNPavel) [inline test/e2e/memcached_test.go]: Why was this moved before the `dep ensure`? `dep ensure` will just remove these symlinks when it runs. This should be done immediately after the `dep ensure`

**Comment by AlexNPavel** [comment]
I guess it's fine to leave the constants as they are. As you said, they shouldn't change often. And other solutions are a bit cumbersome.

**Comment by joelanford** [comment]
Just a thought, but #756 is potentially related to this issue. If we moved `pkg/scaffold` into `internal`, would that solve the problem?

**Comment by estroz** [comment]
@joelanford yes. We can have `projutil` functions required by `pkg/scaffold` in the `scaffold` package so we don't need to import constants. Also none of the stuff in `yamlutil/manifests.go` is used by scaffolds.
